addons/velero/1.6.2/install.sh

```diff
@@ -1,34 +1,63 @@
-
+# shellcheck disable=SC2148
 function velero_pre_init() {
     if [ -z "$VELERO_NAMESPACE" ]; then
         VELERO_NAMESPACE=velero
     fi
     if [ -z "$VELERO_LOCAL_BUCKET" ]; then
         VELERO_LOCAL_BUCKET=velero
     fi
+    # TODO (dans): make this configurable from the installer spec
+    # if [ -z "$VELERO_REQUESTED_CLAIM_SIZE" ]; then
+    #     VELERO_REQUESTED_CLAIM_SIZE="50Gi"
+    # fi
 }
 
+# runs on first install, and on version upgrades only
 function velero() {
     local src="$DIR/addons/velero/$VELERO_VERSION"
     local dst="$DIR/kustomize/velero"
 
-    cp "$src/kustomization.yaml" "$dst/"
+    render_yaml_file "$src/tmpl-kustomization.yaml" > "$dst/kustomization.yaml"
 
     velero_binary
 
+    determine_velero_pvc_size
+
     velero_install "$src" "$dst"
 
     velero_patch_restic_privilege "$src" "$dst"
 
     velero_kotsadm_restore_config "$src" "$dst"
 
+    # always patch the velero and restic manifests to include the PVC case
+    velero_patch_internal_pvc_snapshots "$src" "$dst"
+
     velero_patch_http_proxy "$src" "$dst"
 
     velero_change_storageclass "$src" "$dst"
 
+    velero_migrate_from_object_store "$src" "$dst"
+
     kubectl apply -k "$dst"
 
     kubectl label -n default --overwrite service/kubernetes velero.io/exclude-from-backup=true
+
+    # Bail if the migratoin fails, preventing the original object store from being deleted
+    if [ "$WILL_MIGRATE_VELERO_OBJECT_STORE" = "1" ]; then
+        logWarn "Velero will migrate from object store to pvc"
+        try_5m velero_pvc_migrated
+        logSuccess "Velero migration complete"
+    fi
+
+    # Patch snapshots volumes to "Retain" in case of deletion
+    if kubernetes_resource_exists "$VELERO_NAMESPACE" pvc velero-internal-snapshots; then
+
+        local velero_pv_name
+        echo "Patching internal snapshot volume Reclaim Policy to RECLAIM"
+        try_1m velero_pvc_bound
+        velero_pv_name=$(kubectl get pvc velero-internal-snapshots -n ${VELERO_NAMESPACE} -ojsonpath='{.spec.volumeName}')
```

Comment:
is the pvc name velero-internal-snapshots, or snapshot-pvc

Reply:
Should be `velero-internal-snapshots`, or at least that the name I picked. I can change it if there was something different you used.

Reply:
I was using `velero-internal-snapshots` but I'm open to changing. I do think "internal" should be part of the name thought to describe where it is used.

Reply:
let us keep the way you wrote it. I can modify my code.